presto-main/src/main/java/com/facebook/presto/operator/ExchangeClient.java

```diff
@@ -90,6 +92,8 @@
     private final AtomicBoolean closed = new AtomicBoolean();
     private final AtomicReference<Throwable> failure = new AtomicReference<>();
 
+    private SystemMemoryUsageListener systemMemoryUsageListener = deltaMemoryInBytes -> { };
```

Comment:
Make this final, and remove the setter. Instead it can be a constructor parameter. Just refactor `ExchangeClientFactory` to implement an interface `ExchangeClientSupplier` which has a method `get(SystemMemoryUsageListener listener)`